tap_mssql/connection.py

```diff
@@ -54,3 +54,11 @@ def __init__(self, *args, **kwargs):
             connect_with_backoff(self)
 
     return ConnectionWrapper
+
+def ResultIterator(cursor, arraysize=1):
+    while True:
+        results = cursor.fetchmany(arraysize)
+        if not results:
+            break
+        for result in results:
+            yield result
```

Comment:
The new `ResultIterator` function allowing tuning on arraysize by extraction of a given number of rows to memory.